### PR TITLE
[JENKINS-71186] - Avoid a warning from web application scanners

### PR DESCRIPTION
**Pull request by programbeginnerTW**
Add X-Content-Type-Options header to avoid web application scanner to flag a potential vulnerability

See [JENKINS-71186](https://issues.jenkins.io/browse/JENKINS-71186).

### Testing done

- Tested interactively

### Proposed changelog entries

-  Add the `X-Content-Type-Options` HTTP header to the response from the agent listener.  Silence security scanners that incorrectly report an issue when the HTTP header is missing.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] There is automated testing or an explanation as to why this change has no tests.
```

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8288"></a>

### CODE REVIEW

**Review by Wadeck — COMMENTED** [review]
It does not correct a vulnerability but if it allows scanners to be less noisy, why not :-)

**Review by MarkEWaite — APPROVED** [review]

**Comment by programbeginnerTW** [comment]
Thanks for the approval ! @Wadeck & @MarkEWaite !
Since I'm not familiar to GitHub,  I'm quite confused why the merge is blocked...
Is it telling me to change the branch to merge? If so, I'll revise it soon.

**Comment by lemeurherve** [comment]
@programbeginnerTW you don't have anything to do, there is a code freeze in this repository as a security release is planned to be delivered today, merges will resume after its completion.

**Comment by programbeginnerTW** [comment]
@lemeurherve  Thanks for the explanation!

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
